src/ConnectedMode/SlCoreConnectionAdapter.cs

```diff
@@ -98,6 +103,39 @@ public Task<AdapterResponseWithData<List<OrganizationDisplay>>> GetOrganizations
         });
     }
 
+    public Task<AdapterResponseWithData<ServerProject>> GetServerProjectByKeyAsync(ServerConnection serverConnection, ConnectionInfo connectionInfo, string serverProjectKey)
+    {
+        var failedResponse = new AdapterResponseWithData<ServerProject>(false, null);
+        
+        return threadHandling.RunOnBackgroundThread(async () =>
+        {
+            if (!TryGetConnectionConfigurationSlCoreService(out var connectionConfigurationSlCoreService))
+            {
+                return failedResponse;
+            }
+
+            try
+            {
+                var credentials = (BasicAuthCredentials) serverConnection.Credentials;
+                if (credentials == null)
+                {
+                    return failedResponse;
+                }
+                var credentialsSlCoreFormat = credentials.Password?.Length > 0
+                    ? Either<TokenDto, UsernamePasswordDto>.CreateRight(new UsernamePasswordDto(credentials.UserName, credentials.Password.ToString()))
```

Comment:
Here it should be `.ToUnsecureString()` instead of `ToString()` (that's why the ProjectName was not returned when using username/password binding)